grantPermission.ps1

```diff
@@ -0,0 +1,241 @@
+            }
+        }
+        
+        # If error message empty, fall back on $ex.Exception.Message
+        if ([String]::IsNullOrEmpty($errorMessage.VerboseErrorMessage)) {
+            $errorMessage.VerboseErrorMessage = $ErrorObject.Exception.Message
+        }
+        if ([String]::IsNullOrEmpty($errorMessage.AuditErrorMessage)) {
+            $errorMessage.AuditErrorMessage = $ErrorObject.Exception.Message
+        }
+
+        Write-Output $errorMessage
+    }
+}
+#endregion functions
+
+try {
+    # Get current account
+    try {
+        Write-Verbose "Querying AFAS user where [$($correlationProperty)] = [$($correlationValue)]"
+
+        # Create authorization headers
+        $encodedToken = [System.Convert]::ToBase64String([System.Text.Encoding]::ASCII.GetBytes($($c.Token)))
+        $authValue = "AfasToken $encodedToken"
+        $Headers = @{ Authorization = $authValue }
+        $Headers.Add("IntegrationId", "45963_140664") # Fixed value - Tools4ever Partner Integration ID
+
+        $splatWebRequest = @{
+            Uri             = "$($c.BaseUri)/connectors/$($c.GetConnector)?filterfieldids=$($correlationProperty)&filtervalues=$($correlationValue)&operatortypes=1"
+            Headers         = $headers
+            Method          = 'GET'
+            ContentType     = "application/json;charset=utf-8"
+            UseBasicParsing = $true
+        }
+        $currentAccount = (Invoke-RestMethod @splatWebRequest -Verbose:$false).rows
+
+        if ($null -eq $currentAccount.Gebruiker) {
+            throw "No AFAS user found where [$($correlationProperty)] = [$($correlationValue)]"
+        }
+    }
+    catch {
+        $ex = $PSItem
+        $errorMessage = Get-ErrorMessage -ErrorObject $ex
+
+        Write-Verbose "Error at Line [$($ex.InvocationInfo.ScriptLineNumber)]: $($ex.InvocationInfo.Line). Error: $($errorMessage.VerboseErrorMessage)"
+
+        if ($errorMessage.AuditErrorMessage -Like "*No AFAS user found*") {
+            $auditLogs.Add([PSCustomObject]@{
+                    # Action  = "" # Optional
+                    Message = "No AFAS user found where [$($correlationProperty)] = [$($aRef.Gebruiker)]. Possibly deleted."
+                    IsError = $true
+                })
+        }
+        else {
+            $auditLogs.Add([PSCustomObject]@{
+                    # Action  = "" # Optional
+                    Message = "Error querying AFAS user where [$($correlationProperty)] = [$($aRef.Gebruiker)]. Error Message: $($errorMessage.AuditErrorMessage)"
+                    IsError = $true
+                })
+        }
+    }
+
+    # Grant permission
+    try {
+        $bodyAddPermission = [PSCustomObject]@{
+            'KnUser' = @{
+                'Element' = @{
+                    # Gebruiker
+                    '@UsId'  = $currentAccount.Gebruiker
+                    'Fields' = @{
+                        # Mutatie code
+                        'MtCd'   = $account.MtCd
```

Comment:
$account.MtCd should be 1

Reply:
Permissions need to be further tested before we add this to the connector